src/niweb/apps/noclook/forms/common.py

```diff
@@ -919,30 +922,43 @@ def clean(self):
             first_name = first_name.encode('utf-8')
             last_name  = last_name.encode('utf-8')
 
-        cleaned_data['name'] = '{} {}'.format(first_name, last_name)
+        full_name = '{} {}'.format(first_name, last_name)
+        node_type = NodeType.objects.get(type="Contact")
+
+        if is_create and self.data and NodeHandle.objects.filter(node_name=full_name, node_type=node_type):
```

Comment:
As my other comment states, this is the opposite of what Stefan meant. This is actually how it worked already due to the form being saved by `form_to_unique_node_handle`.

Reply:
My fault, I understood the total opposite :( thankfully the fix was easy, thanks for pointing me out the right helper :)